
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>molecular QTLs</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css?v=6644e6bb" />
    <link rel="stylesheet" type="text/css" href="../_static/plot_directive.css?v=7f9a90b1" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=ad64850d" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-dropdown.css?v=995e94df" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-bootstrap.min.css?v=21c0b90a" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=ad64850d" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=5929fcd5"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'softwares/02.2molQTL';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Integrative analysis" href="04.0IntegrativeAnalysis.html" />
    <link rel="prev" title="LD calculation" href="02.1LDref.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
    <meta name="docbuild:last-update" content="2025-08-01"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search across website"
         aria-label="Search across website"
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class=" navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
  
    <p class="title logo__title"></p>
  
</a></div>
    
  </div>
  
  <div class=" navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item current active">
  <a class="nav-link nav-internal" href="content-softwares.html">
    BayesOmics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../tutorials/content-tutorials.html">
    Tutorials
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../fileFormats/content-fileFormats.html">
    File formats
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../notes/content-notes.html">
    Notes
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../reference/content-reference.html">
    Reference
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
        </div>
      
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/ShouyeLiu" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item current active">
  <a class="nav-link nav-internal" href="content-softwares.html">
    BayesOmics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../tutorials/content-tutorials.html">
    Tutorials
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../fileFormats/content-fileFormats.html">
    File formats
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../notes/content-notes.html">
    Notes
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../reference/content-reference.html">
    Reference
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/ShouyeLiu" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
<nav class="bd-docs-nav bd-links"
     aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">Section Navigation</p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01.0installation.html">Installation</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="02.0dataManagement.html">Data management</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="02.1LDref.html">LD calculation</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">molecular QTLs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="04.0IntegrativeAnalysis.html">Integrative analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="04.1Simulation.html">Simulation</a></li>
<li class="toctree-l2"><a class="reference internal" href="04.1Imputation.html">Imputation and adjustment</a></li>
<li class="toctree-l2"><a class="reference internal" href="04.2PolygenicPrediction.html">Polygenic prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="04.3MediatedHeritability.html">Mediated heritability estimation</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="05.0dataResource.html">Data Resource</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="05.2LDandMolQTLs.html">LD and molecular QTLs</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="06.0questions.html">FAQs</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">



<nav aria-label="Breadcrumb" class="d-print-none">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="../index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    
    <li class="breadcrumb-item"><a href="content-softwares.html" class="nav-link">BayesOmics</a></li>
    
    
    <li class="breadcrumb-item"><a href="02.0dataManagement.html" class="nav-link">Data management</a></li>
    
    <li class="breadcrumb-item active" aria-current="page">molecular QTLs</li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="molecular-qtls">
<h1>molecular QTLs<a class="headerlink" href="#molecular-qtls" title="Link to this heading">#</a></h1>
<section id="molqtl-summary-statistics">
<span id="xqtldatmanagement"></span><h2>molQTL summary statistics<a class="headerlink" href="#molqtl-summary-statistics" title="Link to this heading">#</a></h2>
<section id="options">
<h3>Options<a class="headerlink" href="#options" title="Link to this heading">#</a></h3>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-eqtl-flist">
<span class="sig-name descname"><span class="pre">--eqtl-flist</span></span><span class="sig-prename descclassname"> <span class="pre">flistfile</span></span><a class="headerlink" href="#cmdoption-eqtl-flist" title="Link to this definition">#</a></dt>
<dd></dd></dl>

<div class="line-block">
<div class="line">read flist eQTL format</div>
</div>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-add-gene-n">
<span class="sig-name descname"><span class="pre">--add-gene-n</span></span><span class="sig-prename descclassname"> <span class="pre">geneSampleSizeFile</span></span><a class="headerlink" href="#cmdoption-add-gene-n" title="Link to this definition">#</a></dt>
<dd></dd></dl>

<div class="line-block">
<div class="line">Add specific sample size for each gene-snp pair</div>
</div>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-keep-gene">
<span class="sig-name descname"><span class="pre">--keep-gene</span></span><span class="sig-prename descclassname"></span><a class="headerlink" href="#cmdoption-keep-gene" title="Link to this definition">#</a></dt>
<dd></dd></dl>

<div class="line-block">
<div class="line">extract genes from files.</div>
</div>
</section>
<section id="convertion-between-smr-besd-format-and-gz-format">
<span id="xqtldatformat"></span><h3>Convertion between SMR BESD format and gz format<a class="headerlink" href="#convertion-between-smr-besd-format-and-gz-format" title="Link to this heading">#</a></h3>
<p><strong>Convert plist format to besd format</strong></p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--eqtl-flist<span class="w"> </span>flistFile<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--make-besd<span class="w"> </span>--out<span class="w"> </span>mybesd
</pre></div>
</div>
<p>Since BESD format can save effect size and se only, a compressed gz format has been designed to incorporate more useful information</p>
<p>psuedo-1kg-chr22-causal-model.q.query.gz</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>GeneID<span class="w">             </span>GeneChr<span class="w">  </span>GeneLength<span class="w">   </span>GenePhyPos<span class="w">   </span>SNPID<span class="w">     </span>SNPChr<span class="w">   </span>SNPPhyPos<span class="w">  </span>A1<span class="w">  </span>A2<span class="w">  </span>A1Freq<span class="w">    </span>BETA<span class="w">        </span>SE<span class="w">   </span>Pvalue<span class="w"> </span>N
ENSG00000070413<span class="w">       </span><span class="m">22</span><span class="w">     </span><span class="m">2000000</span><span class="w">     </span><span class="m">19066881</span>.0<span class="w">   </span>rs2073776<span class="w">   </span><span class="m">22</span><span class="w">    </span><span class="m">19024651</span><span class="w">    </span>C<span class="w">   </span>T<span class="w">     </span><span class="m">0</span>.499<span class="w">   </span><span class="m">0</span>.0015<span class="w">  </span><span class="m">0</span>.0044<span class="w">  </span><span class="m">0</span>.72<span class="w">    </span><span class="m">100000</span>
ENSG00000070413<span class="w">       </span><span class="m">22</span><span class="w">     </span><span class="m">2000000</span><span class="w">     </span><span class="m">19066881</span>.0<span class="w">   </span>rs6623<span class="w">      </span><span class="m">22</span><span class="w">    </span><span class="m">19025381</span><span class="w">    </span>C<span class="w">   </span>T<span class="w">     </span><span class="m">0</span>.499<span class="w">   </span><span class="m">0</span>.0016<span class="w">  </span><span class="m">0</span>.0044<span class="w">  </span><span class="m">0</span>.70<span class="w">    </span><span class="m">100000</span>
ENSG00000070413<span class="w">       </span><span class="m">22</span><span class="w">     </span><span class="m">2000000</span><span class="w">     </span><span class="m">19066881</span>.0<span class="w">   </span>rs10160<span class="w">     </span><span class="m">22</span><span class="w">    </span><span class="m">19026025</span><span class="w">    </span>G<span class="w">   </span>A<span class="w">     </span><span class="m">0</span>.498<span class="w">   </span>-0.003<span class="w">  </span><span class="m">0</span>.0044<span class="w">  </span><span class="m">0</span>.44<span class="w">    </span><span class="m">100000</span>
......
</pre></div>
</div>
<p>So we can convert BESD format to query.gz format if additional gene sample size is needed</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--beqtl-summary<span class="w"> </span>mybesd<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--make-query<span class="w">  </span>--out<span class="w"> </span>myquery
</pre></div>
</div>
<p>Please see <a class="reference external" href="https://yanglab.westlake.edu.cn/software/smr/#MakeaBESDfile">SMR flist to besd</a> for details.</p>
<p><strong>Extract a single genes basd on gene input file</strong></p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--beqtl-summary<span class="w"> </span>mybesd<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--keep-one-gene<span class="w"> </span>geneID<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--make-query<span class="w"> </span>--out<span class="w"> </span>myquery
</pre></div>
</div>
<p><strong>Add gene sample size</strong></p>
<p>There are two ways to add sample size into analysis:</p>
<ol class="arabic simple">
<li><p>when all SNPs within the same gene have the same sample size,</p></li>
</ol>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>GeneID<span class="w">        </span>N
ENSG00000182902<span class="w">       </span><span class="m">10000</span>
ENSG00000183628<span class="w">       </span><span class="m">10000</span>
ENSG00000100346<span class="w">       </span><span class="m">10000</span>
......
</pre></div>
</div>
<p>we suggest that adding sample size information into BESD format directly by the folloing command:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--beqtl-summary<span class="w"> </span>mybesd<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--add-gene-n<span class="w"> </span>geneSampleSizeFile<span class="se">\</span>
<span class="w">    </span>--make-query<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--out<span class="w"> </span>myquery
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>when SNPs with the same gene may have different sample size (usually occur in meta analysis, like eQTLs from eQTLGen project),</p></li>
</ol>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>GeneID<span class="w">        </span>N
ENSG00000182902<span class="w">       </span><span class="m">10000</span>
ENSG00000183628<span class="w">       </span><span class="m">10000</span>
ENSG00000100346<span class="w">       </span><span class="m">10000</span>
......
</pre></div>
</div>
<p>we suggest that adding these sample size into gz format by the following command:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--beqtl-summary<span class="w"> </span>mybesd<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--add-gene-n<span class="w"> </span>geneSampleSizeFile<span class="se">\</span>
<span class="w">    </span>--make-query<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--out<span class="w"> </span>myquery
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>A)</strong> If you’re adding gene sample size to the analysis, the original sample size in the BESD or gz file will be replaced and gene-snp pairs with NA sample size will be deleted automatically. <strong>B)</strong> If per gene sample size is missing, you can also add total sample size to <a class="reference external" href="https://yanglab.westlake.edu.cn/software/smr/#BESDformat">besd format</a> file by using <a class="reference external" href="https://yanglab.westlake.edu.cn/software/smr/#Mangethesamplesize">SMR</a> software, and we will use total sample size for each gene. <strong>C)</strong> If both total and per gene sample size are missing, we will use estimated gene sample size in the analysis (add this function later).</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Since the LD matrix is the low-rank LD, it’s important to ensure that the gene and SNP set in the <strong>geneSampleSizeFile</strong> are identical to those in the BESD or gz file. If they’re not identical, you’ll need to perform QC first to ensure they match first.</p>
</div>
<p><strong>Merge multiple besd file into one*</strong></p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>BayesOmics64<span class="w"> </span>--beqtl-list<span class="w"> </span>mybesdListFile.txt<span class="w"> </span>--merge-besd<span class="w"> </span>--out<span class="w"> </span>mergeBesd
</pre></div>
</div>
<p>Input file format
mybesdListFile.txt (full paths can be specified if the besd files are in different directories)</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>test_1
test_2
test_3
......
test_22
</pre></div>
</div>
</section>
<section id="qc-for-molqtls">
<h3>QC for molQTLs<a class="headerlink" href="#qc-for-molqtls" title="Link to this heading">#</a></h3>
<p>It will very convenient to do manipulations on molQTL datasets using R, as the following example:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="n">eqtl</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;example.query.gz&quot;</span><span class="p">))</span>
<span class="c1">####################################</span>
<span class="n">cisWin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000000</span>
<span class="n">dat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eqtl</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">group_by</span><span class="p">(</span><span class="n">GeneChr</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w"> </span><span class="n">cisStart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">pmax</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">GenePhyPos</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">cisWin</span><span class="p">),</span>
<span class="w"> </span><span class="n">cisEnd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GenePhyPos</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">cisWin</span>
<span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="nf">filter</span><span class="p">(</span><span class="n">GeneChr</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">SNPChr</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="c1"># Ensure SNP and gene are on the same chromosome</span>
<span class="nf">filter</span><span class="p">(</span><span class="n">SNPPhyPos</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="n">cisStart</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">SNPPhyPos</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="n">cisEnd</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="nf">filter</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">BETA</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1e-6</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="nf">select</span><span class="p">(</span><span class="nf">all_of</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">eqtl</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="n">GeneChr</span><span class="p">,</span><span class="n">GenePhyPos</span><span class="p">,</span><span class="w"> </span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">SNPPhyPos</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="02.1LDref.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">LD calculation</p>
      </div>
    </a>
    <a class="right-next"
       href="04.0IntegrativeAnalysis.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Integrative analysis</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
<div
    id="pst-page-navigation-heading-2"
    class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc" aria-labelledby="pst-page-navigation-heading-2">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#molqtl-summary-statistics">molQTL summary statistics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#options">Options</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#convertion-between-smr-besd-format-and-gz-format">Convertion between SMR BESD format and gz format</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#qc-for-molqtls">QC for molQTLs</a></li>
</ul>
</li>
</ul>
  </nav></div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/ShouyeLiu/ShouyeLiu/edit/master/docs/source/softwares/02.2molQTL.rst">
      <i class="fa-solid fa-pencil"></i>
      
      
        
          Edit on GitHub
        
      
    </a>
  </div>
</div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

<!-- your html code here -->
  <!-- <style> -->
  <!--   /* Sidebar header (and topbar for mobile) */ -->
  <!--   .wy-side-nav-search, .wy-nav-top { -->
  <!--     background: #00ff00; -->
  <!--   } -->
  <!--   /* Sidebar */ -->
  <!--   .wy-nav-side { -->
  <!--     background: #ff0000; -->
  <!--   } -->
  <!-- </style> -->

  </body>
</html>