
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Imputation and adjustment</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/graphviz.css?v=fd3f3429" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/plot_directive.css?v=7f9a90b1" />
    <link rel="stylesheet" type="text/css" href="../_static/jupyter-sphinx.css?v=572af1d6" />
    <link rel="stylesheet" type="text/css" href="../_static/thebelab.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=ad64850d" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-dropdown.css?v=995e94df" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-bootstrap.min.css?v=21c0b90a" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=ad64850d" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=7f41d439"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/thebelab-helper.js"></script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/@jupyter-widgets/html-manager@^1.0.1/dist/embed-amd.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'softwares/04.1Imputation';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Polygenic prediction" href="04.2PolygenicPrediction.html" />
    <link rel="prev" title="Integrative analysis" href="04.0IntegrativeAnalysis.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
    <meta name="docbuild:last-update" content="2025-05-27"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search across website"
         aria-label="Search across website"
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class=" navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
  
    <p class="title logo__title"></p>
  
</a></div>
    
  </div>
  
  <div class=" navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item current active">
  <a class="nav-link nav-internal" href="content-softwares.html">
    BayesOmics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../fileFormats/content-fileFormats.html">
    File formats
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../notes/content-notes.html">
    Notes
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../reference/content-reference.html">
    Reference
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
        </div>
      
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/ShouyeLiu" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item current active">
  <a class="nav-link nav-internal" href="content-softwares.html">
    BayesOmics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../fileFormats/content-fileFormats.html">
    File formats
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../notes/content-notes.html">
    Notes
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="../reference/content-reference.html">
    Reference
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/ShouyeLiu" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
<nav class="bd-docs-nav bd-links"
     aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">Section Navigation</p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01.0installation.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="02.0dataManagement.html">Data management</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="02.1LDref.html">LD calculation</a></li>
<li class="toctree-l2"><a class="reference internal" href="02.2molQTL.html">molecular QTLs</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="04.0IntegrativeAnalysis.html">Integrative analysis</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Imputation and adjustment</a></li>
<li class="toctree-l2"><a class="reference internal" href="04.2PolygenicPrediction.html">Polygenic prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="04.3MediatedHeritability.html">Mediated heritability estimation</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="05.0dataResource.html">Data Resource</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="05.2LDandMolQTLs.html">LD and molecular QTLs</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="06.0questions.html">FAQs</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">



<nav aria-label="Breadcrumb" class="d-print-none">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="../index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    
    <li class="breadcrumb-item"><a href="content-softwares.html" class="nav-link">BayesOmics</a></li>
    
    
    <li class="breadcrumb-item"><a href="04.0IntegrativeAnalysis.html" class="nav-link">Integrative analysis</a></li>
    
    <li class="breadcrumb-item active" aria-current="page">Imputation...</li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="imputation-and-adjustment">
<h1>Imputation and adjustment<a class="headerlink" href="#imputation-and-adjustment" title="Link to this heading">#</a></h1>
<section id="theory-underlying-the-adjustment-of-sample-sizes-in-gwas-molqtl-summary-statistics">
<h2>Theory underlying the adjustment of sample sizes in GWAS (molQTL) summary statistics<a class="headerlink" href="#theory-underlying-the-adjustment-of-sample-sizes-in-gwas-molqtl-summary-statistics" title="Link to this heading">#</a></h2>
<p>For a given gene (or protein), we defined <span class="math notranslate nohighlight">\(b_j^{(c)}\)</span> as the molQTL effect for SNP <span class="math notranslate nohighlight">\(j\)</span> with the current sample size and <span class="math notranslate nohighlight">\(b_j^{(t)}\)</span> as that with the target sample size. Under the assumption of normality and unit phenotypic variance, the joint distribution of <span class="math notranslate nohighlight">\(b_j^{(c)}\)</span> and <span class="math notranslate nohighlight">\(b_j^{(t)}\)</span> can be approximated by a bivariate normal distribution.</p>
<p>For each gene, given the marginal effect-size vector <span class="math notranslate nohighlight">\(\mathbf{b}^{(c)}\)</span> and LD matrix <span class="math notranslate nohighlight">\(\mathbf{R}\)</span>, the conditional distribution of <span class="math notranslate nohighlight">\(\mathbf{b}^{(t)} \mid \mathbf{b}^{(c)}\)</span> is:</p>
<div class="math notranslate nohighlight">
\[\mathbf{b}^{(t)} \mid \mathbf{b}^{(c)} \sim \mathcal{N}\left(\mathbf{b}^{(c)}, \left(\frac{1}{n_t} - \frac{1}{n_c} \right)\mathbf{R} \right)\]</div>
<p>where <span class="math notranslate nohighlight">\(n_c\)</span> and <span class="math notranslate nohighlight">\(n_t\)</span> are the current and target sample sizes for each molQTL within that gene, respectively, with <span class="math notranslate nohighlight">\(\mathbf{b}^{(c)}\)</span> used as a surrogate of the true effect sizes.</p>
<p>Eigen-decomposition with a maximum cut-off of variance explained (<span class="math notranslate nohighlight">\(\rho = 0.995\)</span>) was applied to the LD matrix to generate a pseudo-inverse of <span class="math notranslate nohighlight">\(\mathbf{R}\)</span>. The summary statistics under the target sample size were then estimated as:</p>
<div class="math notranslate nohighlight">
\[\mathbf{b}^{(t)} \mid \mathbf{b}^{(c)} = \mathbf{b}^{(c)} + \sqrt{\frac{1}{n_t} - \frac{1}{n_c}} \, \mathbf{U} \mathbf{\Lambda}^{1/2} \boldsymbol{\xi}\]</div>
<p>where <span class="math notranslate nohighlight">\(\mathbf{U}\)</span> and <span class="math notranslate nohighlight">\(\mathbf{\Lambda}\)</span> are matrices of eigenvectors and eigenvalues of the LD matrix <span class="math notranslate nohighlight">\(\mathbf{R}\)</span>, and <span class="math notranslate nohighlight">\(\xi_j \sim \mathcal{N}(0, 1)\)</span> is a standard normal random variable for SNP <span class="math notranslate nohighlight">\(j\)</span>.</p>
<p>The marginal standard errors (SE) for <span class="math notranslate nohighlight">\(b^{(t)}\)</span> and <span class="math notranslate nohighlight">\(b^{(c)}\)</span> are:</p>
<div class="math notranslate nohighlight">
\[\mathrm{SE}^{(t)} = \sqrt{\frac{1 - \left(b^{(t)}\right)^2}{n_t}}, \quad \mathrm{SE}^{(c)} = \sqrt{\frac{1 - \left(b^{(c)}\right)^2}{n_c}}\]</div>
<p>Equivalently, <span class="math notranslate nohighlight">\(\mathrm{SE}^{(t)}\)</span> can be estimated as:</p>
<div class="math notranslate nohighlight">
\[\mathrm{SE}^{(t)} = \sqrt{\frac{n_c}{n_t}} \, \mathrm{SE}^{(c)}\]</div>
</section>
<section id="implementation-of-method-to-adjust-sample-sizes-in-gwas-molqtl-summary">
<h2>Implementation of method to adjust sample sizes in GWAS (molQTL) summary<a class="headerlink" href="#implementation-of-method-to-adjust-sample-sizes-in-gwas-molqtl-summary" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#############################################################</span>
<span class="c1">### R demo for adjusting sample sizes in GWAS (molQTL) summary</span>
<span class="c1">#############################################################</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ls</span><span class="p">())</span>
<span class="nf">suppressPackageStartupMessages</span><span class="p">({</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">latex2exp</span><span class="p">)</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
<span class="p">})</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">1234</span><span class="p">)</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;SBayesOmics&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">devtools</span><span class="o">::</span><span class="nf">install_github</span><span class="p">(</span><span class="s">&quot;ShouyeLiu/SBayesOmics&quot;</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">library</span><span class="p">(</span><span class="n">SBayesOmics</span><span class="p">)</span>
<span class="n">sumPath</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata/summary/&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">package</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SBayesOmics&quot;</span><span class="p">)</span>

<span class="n">smrIndFileSuffix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ukbg-chr22-causal-model-50k-ind&quot;</span>
<span class="n">geneLDFile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">sumPath</span><span class="p">,</span><span class="n">smrIndFileSuffix</span><span class="p">)</span>
<span class="n">geneSummaryFile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">sumPath</span><span class="p">,</span><span class="n">smrIndFileSuffix</span><span class="p">)</span>
<span class="n">qtlP50K</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">downsampleN</span><span class="p">(</span><span class="n">geneSummaryFile</span><span class="p">,</span><span class="n">geneLDFile</span><span class="p">,</span><span class="n">targetN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50000</span><span class="p">)</span>
<span class="n">qtlP10K</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">downsampleN</span><span class="p">(</span><span class="n">geneSummaryFile</span><span class="p">,</span><span class="n">geneLDFile</span><span class="p">,</span><span class="n">targetN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10000</span><span class="p">)</span>
<span class="n">qtlP1K</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="nf">downsampleN</span><span class="p">(</span><span class="n">geneSummaryFile</span><span class="p">,</span><span class="n">geneLDFile</span><span class="p">,</span><span class="n">targetN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span>
<span class="n">datPse</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">qtlP50K</span><span class="p">,</span><span class="n">qtlP10K</span><span class="p">,</span><span class="n">qtlP1K</span><span class="p">)</span>
<span class="c1">############################################################</span>
<span class="c1">## specific theme</span>
<span class="n">themeLocal</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">theme_bw</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w"> </span><span class="n">axis.title</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="n">face</span><span class="o">=</span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w"> </span><span class="n">axis.title.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8.5</span><span class="p">),</span><span class="w">  </span><span class="c1"># Adjust the size as needed</span>
<span class="w"> </span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8.5</span><span class="p">),</span>
<span class="w"> </span><span class="n">legend.title</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="n">face</span><span class="o">=</span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w"> </span><span class="n">legend.text</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w"> </span><span class="n">panel.grid.major</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">),</span>
<span class="w"> </span><span class="n">panel.grid.minor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.1</span><span class="p">),</span>
<span class="w"> </span><span class="n">plot.subtitle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">),</span>
<span class="w"> </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5.5</span><span class="p">),</span>
<span class="w"> </span><span class="n">aspect.ratio</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="p">,</span>
<span class="w"> </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">),</span>
<span class="w"> </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">),</span>
<span class="w"> </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0.65</span><span class="p">,</span><span class="w"> </span><span class="m">0.12</span><span class="p">),</span>
<span class="w"> </span><span class="n">legend.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">alpha</span><span class="p">(</span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="w"> </span><span class="p">)</span>
<span class="n">datPse</span><span class="o">$</span><span class="n">N</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">datPse</span><span class="o">$</span><span class="n">N</span><span class="p">)</span>
<span class="n">mycolors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#45b787&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;#f03f24&quot;</span><span class="p">,</span><span class="s">&quot;black&quot;</span><span class="p">)</span>
<span class="c1">##########################################</span>
<span class="n">pMBeta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">datPse</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="nf">as.data.table</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">betaR</span><span class="p">,</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">betaPse</span><span class="p">,</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">N</span><span class="p">,</span><span class="n">shape</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">N</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">geom_abline</span><span class="p">(</span><span class="n">intercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">slope</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="n">stroke</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">  </span><span class="o">+</span>
<span class="w"> </span><span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;True eQTL effect&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;Estimated eQTL effect&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="n">mycolors</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="n">mycolors</span><span class="p">)</span><span class="o">+</span>
<span class="w"> </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">GeneID</span><span class="p">,</span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="n">themeLocal</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="c1"># + theme(legend.position = &quot;none&quot;) +</span>
<span class="w"> </span><span class="nf">labs</span><span class="p">(</span><span class="n">color</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Sample size&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">shape</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Sample size&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">labs</span><span class="p">(</span><span class="n">tag</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;a&#39;</span><span class="p">)</span>
<span class="n">pMSe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">datPse</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="nf">as.data.table</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seR</span><span class="p">,</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sePse</span><span class="p">,</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">N</span><span class="p">,</span><span class="n">shape</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">N</span><span class="w"> </span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">geom_abline</span><span class="p">(</span><span class="n">intercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">slope</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">geom_point</span><span class="p">(</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="n">stroke</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">  </span><span class="o">+</span>
<span class="w"> </span><span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;True eQTL standard error (SE)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;Estimated eQTL standard error (SE)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="n">mycolors</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="n">mycolors</span><span class="p">)</span><span class="o">+</span>
<span class="w"> </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">GeneID</span><span class="p">,</span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="n">themeLocal</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="nf">labs</span><span class="p">(</span><span class="n">color</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Sample size&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">shape</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Sample size&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w"> </span><span class="c1"># guides(color = guide_legend(title = &quot;Sample size&quot;)) +</span>
<span class="w"> </span><span class="nf">labs</span><span class="p">(</span><span class="n">tag</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;b&#39;</span><span class="p">)</span>
<span class="n">fplot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">pMBeta</span><span class="p">,</span><span class="w"> </span><span class="n">pMSe</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">rel_widths</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">))</span>
<span class="c1"># # Show it</span>
<span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w">  </span><span class="c1"># adjust size</span>
<span class="nf">print</span><span class="p">(</span><span class="n">fplot</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>10 gene low-rank marices have been read.
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>10 gene low-rank marices have been read.
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>10 gene low-rank marices have been read.
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
<span class=" -Color -Color-Cyan">ℹ</span> Please use the `linewidth` argument instead.”
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“A numeric `legend.position` argument in `theme()` was deprecated in ggplot2 3.5.0.
<span class=" -Color -Color-Cyan">ℹ</span> Please use the `legend.position.inside` argument of `theme()` instead.”
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/145bd179148687bfbeff681d40e6543e3f27dff76b27b68c94430098f224f867.png"><img alt="../_images/145bd179148687bfbeff681d40e6543e3f27dff76b27b68c94430098f224f867.png" src="../_images/145bd179148687bfbeff681d40e6543e3f27dff76b27b68c94430098f224f867.png" style="width: 600px; height: 300px;" />
</a>
</div>
</div>
</section>
<section id="citation">
<h2>Citation<a class="headerlink" href="#citation" title="Link to this heading">#</a></h2>
<p>[1] Shouye Liu, Xiaoyu Qian, Yang Wu, Zhili Zheng, Michael E. Goddard, Jian Yang, Peter M. Visscher, Jian Zeng. <a class="reference external" href="https://www.medrxiv.org/content/10.1101/2025.05.24.25328265v1"><strong>Transcriptional and translational genetic control in blood contributes significantly to complex trait heritability and polygenic prediction</strong></a>. Preprint available on bioRxiv doi: https://doi.org/10.1101/2025.05.24.25328265</p>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="04.0IntegrativeAnalysis.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Integrative analysis</p>
      </div>
    </a>
    <a class="right-next"
       href="04.2PolygenicPrediction.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Polygenic prediction</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
<div
    id="pst-page-navigation-heading-2"
    class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc" aria-labelledby="pst-page-navigation-heading-2">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#theory-underlying-the-adjustment-of-sample-sizes-in-gwas-molqtl-summary-statistics">Theory underlying the adjustment of sample sizes in GWAS (molQTL) summary statistics</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation-of-method-to-adjust-sample-sizes-in-gwas-molqtl-summary">Implementation of method to adjust sample sizes in GWAS (molQTL) summary</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#citation">Citation</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

<!-- your html code here -->
  <!-- <style> -->
  <!--   /* Sidebar header (and topbar for mobile) */ -->
  <!--   .wy-side-nav-search, .wy-nav-top { -->
  <!--     background: #00ff00; -->
  <!--   } -->
  <!--   /* Sidebar */ -->
  <!--   .wy-nav-side { -->
  <!--     background: #ff0000; -->
  <!--   } -->
  <!-- </style> -->

  </body>
</html>